books/protocol/src/pruning.md

```diff
@@ -0,0 +1,374 @@
+}
+```
+
+#### explaining what the function is doing in depth:
+
+```c++
+const uint32_t stripe = get_pruning_stripe(pruning_seed);
+if (stripe == 0)
+  return blockchain_height;
+if (block_height + CRYPTONOTE_PRUNING_TIP_BLOCKS >= blockchain_height)
+  return blockchain_height;
+```
+
+This is calculating the [stripe](#getting-a-seeds-pruning-stripe) of the inputted pruning seed, remember if the seed/stripe is `0` that means no pruning so we can return the blockchain height as the next un-pruned height and similarly if the blocks height is within [Tip Blocks](#tip-blocks) of the blockchains height that also means the block won't be pruned.
+
+Returning the blockchains height means the next pruned block doesn't currently exist, its bigger than or equal to blockchain_height - CRYPTONOTE_PRUNING_TIP_BLOCKS or it means it
+will never exist in the case of a zero pruning seed.
+
+```c++
+const uint32_t seed_log_stripes = get_pruning_log_stripes(pruning_seed);
+const uint64_t log_stripes = seed_log_stripes ? seed_log_stripes : CRYPTONOTE_PRUNING_LOG_STRIPES;
+const uint64_t mask = (1ul << log_stripes) - 1;
+```
+
+This is calculating the [log stripes](#getting-a-seeds-log-stripes) of the seed, although Monero currently only allows a log stripes of 3 in the future a higher number could be allowed so this function accounts for that.
+
+If the seeds log stripes is zero this will set it to `CRYPTONOTE_PRUNING_LOG_STRIPES` which is currently `3`.
+
+Then this sets the value of `mask` to one less than the amount of [stripes](#stripes), for Monero the amount of stripes is 8 so `mask` will be 7.
+
+```c++
+const uint32_t block_pruning_seed = ((block_height / CRYPTONOTE_PRUNING_STRIPE_SIZE) & mask) + 1;
+if (block_pruning_seed != stripe)
+  return block_height;
+```
+
+> There is a typo here it should be block_pruning_stripe, think of this as foreshadowing what we are about to do
+
+This calculates the [blocks pruning ~~seed~~ STRIPE](#getting-a-blocks-pruning-stripe) using the same method that we saw in [this](#getting-a-blocks-pruning-stripe) function.
+
+This then checks if the blocks stripe is NOT the same as the seed stripe, if you remember if a seed and block don't have the same stripe that means the seed will prune that block, so we can just return the entered `block_height`.
+
+```c++
+const uint32_t next_stripe = 1 + (block_pruning_seed & mask);
+return get_next_unpruned_block_height(block_height, blockchain_height, tools::make_pruning_seed(next_stripe, log_stripes));
+```
+
+Because the seeds stripe == the blocks stripe we need to work out when our stripe ends/
+when the next stripe starts to get the next pruned block. To do this we can simply calculate the next stripe, make a [new pruning seed](#generating-pruning-seeds) and pass in that seed, which has a stripe one more than ours, into [get next un-pruned block](#getting-the-next-unpruned-block) to get the start of the next stripes un-pruned set and therefore the start of our next pruned set.
```

Comment:
```suggestion
Because the seed's stripe == the block's stripe we need to work out when our stripe ends (when the next stripe starts to get the next pruned block). To do this we can simply calculate the next stripe, make a [new pruning seed](#generating-pruning-seeds) and pass in that seed, which has a stripe one more than ours, into [get next un-pruned block](#getting-the-next-unpruned-block) to get the start of the next stripe's un-pruned set and therefore the start of our next pruned set.
```